test/tests/searchTest.js

```diff
@@ -83,4 +83,58 @@ describe("Zotero.Search", function() {
 			assert.propertyVal(conditions[0], 'value', 'foo')
 		});
 	});
+
+	describe("#search()", function () {
+		let win;
+	
+		before(function () {
+			// Hidden browser, which requires a browser window, needed for charset detection
+			// (until we figure out a better way)
+			if (!Zotero.isStandalone) {
+				return loadBrowserWindow().then(window => win = window);
```

Comment:
I know this wasn't your code to begin with, but better to make the `before` handle a generator and use `win = yield loadBrowserWindow();` here.